packages/eslint-plugin/src/rules/class-literal-property-style.ts

```diff
@@ -89,16 +89,28 @@ export default createRule<Options, MessageIds>({
             return;
           }
 
+          const sourceCode = getSourceCode(context);
+          const name = sourceCode.getText(node.key);
+
+          if (node.parent.type === AST_NODE_TYPES.ClassBody) {
```

Comment:
```suggestion
          if (node.parent.type === AST_NODE_TYPES.ClassBody) {
```
Is it possible that the parent of a MethodDefinition is not a ClassBody? 🤔

Reply:
I think it's impossible. MethodDefinition is only used in ClassElement, which is only used in ClassBody
+ I couldn't find a snapshot in `ast-spec` tests where MethodDefinition is used elsewhere than in ClassBody

Reply:
Ooh, this is a good example of what will be caught by #6225. Linking so we don't miss it.